### PR TITLE
[Rpi3 backport of #74] tee: add support for session's client UUID generation

### PR DESCRIPTION
**Pull request by jforissier**
Cherry-pick two patches from the optee branch corresponding to #74.
Untested.

### CODE REVIEW

File: include/linux/tee_drv.h [hunk 1]
```diff
@@ -153,6 +153,114 @@ int tee_device_register(struct tee_device *teedev);
...
+/**
+ * tee_shm_pool_mgr_destroy() - Free a shared memory manager
+ */
+static inline void tee_shm_pool_mgr_destroy(struct tee_shm_pool_mgr *poolm)
+{
+	poolm->ops->destroy_poolmgr(poolm);
+}
+
+/**
+>>>>>>> baa151f2924f... tee: add support for session's client UUID generation
```
Comment (jenswikl) [inline include/linux/tee_drv.h]: Something left from a merge conflict.

Comment (jforissier) [reply include/linux/tee_drv.h]: Oops! Fixed.

**Comment by jenswi-linaro** [comment]
Looks good to me.

**Comment by jforissier** [comment]
Cherry-picked commit e2aca5d8928a ("tee: flexible shared memory pool creation") from upstream to resolve a build issue. Compile tested only.

**Pull request merged**